src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -257,6 +266,19 @@ public Instance(
         } else {
             log.info("collect_default_jvm_metrics is false - not collecting default JVM metrics");
         }
+        //create bean to track data of this instance
+        JMXbean = beanManager.createJMXBean(this);
+    }
+
+    public void updateBeanInfo()throws Exception {
+        if (JMXbean != null){
+            refreshBeansList();
```

Comment:
This and the below `getMatchingAttributes` are expensive operations, if the instance is monitoring a JVM exposing 10s of thousands of beans, then we must go fetch each of those 10s of thousands of beans and check each attribute to see if it must be monitored.

In the current draft of this PR, we're calling `updateBeanInfo` in the main collect loop every 15s, so this will create a LOT more JMX traffic and use up more CPU.

We should try to structure the updates to the `beanCount` `attributeCount` and `metricCount` to use the existing bean refresh, existing attribute match, and existing metric collection instead of calling them again here.